cdap-proto/src/main/java/co/cask/cdap/proto/QueryRequest.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.proto;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Metrics Query Request format
+ */
+public class QueryRequest {
+  /**
+   * Format for metrics query in batched queries
+   */
+    Map<String, String> tags;
+    List<String> metrics;
+    List<String> groupBy;
+    Map<String, String> timeRange;
```

Comment:
Can we have a real class with optional fields? It can expose proper constructors to allow specify either aggregated or start/stop or start/count. Thoughts?